packages/samples/headless-ssr-commerce/app/search/page.tsx

```diff
@@ -1,3 +1,21 @@
+import {headers} from 'next/headers';
+import SearchPage from '../_components/pages/search-page';
+import {searchEngineDefinition} from '../_lib/commerce-engine';
+import {NextJsNavigatorContext} from '../_lib/navigatorContextProvider';
+
 export default async function Search() {
-  return <h1>Not Supported Yet</h1>;
+  // Sets the navigator context provider to use the newly created `navigatorContext` before fetching the app static state
+  const navigatorContext = new NextJsNavigatorContext(headers());
+  searchEngineDefinition.setNavigatorContextProvider(() => navigatorContext);
```

Comment:
That's something I've been wondering about: why do we do `setNavigatorContextProvider` instead of passing the context provider in the `engineConfig` when we call `defineCommerceEngine`?

@y-lakhdar